mainwindow.cpp

```diff
@@ -1147,8 +1135,16 @@ void MainWindow::setAvailabilityOfFeatures(bool isJumbo)
         QStringList defaultFormats;
         defaultFormats << tr("Auto detect") << "descrypt" << "bsdicrypt" << "md5crypt"
                        << "bcrypt" << "AFS" << "LM" << "crypt" << "tripcode" << "dummy";
+        QString savedFormat = m_ui->comboBox_Format->currentText();
         m_ui->comboBox_Format->clear();
         m_ui->comboBox_Format->addItems(defaultFormats);
+        // Restore user's selection
+        int indexSavedFormat = m_ui->comboBox_Format->findText(savedFormat);
+        if (indexSavedFormat != -1) {
+            m_ui->comboBox_Format->setCurrentIndex(indexSavedFormat);
+        } else if (!savedFormat.isEmpty()){
+            m_ui->comboBox_Format->setEditText(savedFormat);
```

Comment:
Here is the logic. If user has a jumbo build, he'll have permanent type suggestion in the format combo box ex: raw-md5, dynamic_0 etc. If he switches back to a non-jumbo build, we'll clear the comboBox and put back the default formats. The previously selected format will be saved(ex: raw-md5) as the user choice(editable text) to keep UI coherent but it will not stay in the drop down list like permanent formats if modified. In this example, the else if was needed since raw-md5 isn't a default format, if it's a default format like DES, it will set the right index of the drop down list